src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -364,11 +376,219 @@ private static bool IsSafeForReordering(BoundExpression expression, RefKind kind
+                            {
+                                refKindsBuilder = ArrayBuilder<RefKind>.GetInstance(argumentRefKindsOpt.Length);
+                                refKindsBuilder.AddRange(argumentRefKindsOpt);
+                            }
+                        }
+                        Debug.Assert(sideEffectsBuilder != null);
+                        Debug.Assert(!positionsAssignedTemps.IsNull);
+
+                        // Find the last argument needed by the interpolation, and store from the last stored index all the way up to that argument
+                        // to ensure we preserve lexical ordering.
+                        var lastNeededParameter = interpolationData.ArgumentPlaceholders.Aggregate(
+                            BoundInterpolatedStringArgumentPlaceholder.UnspecifiedParameter,
+                            (lastParameter, placeholder) => lastParameter > placeholder.ArgumentIndex ? lastParameter : placeholder.ArgumentIndex);
+                        Debug.Assert(lastNeededParameter >= BoundInterpolatedStringArgumentPlaceholder.InstanceParameter);
+
+                        if (!receiverStored && requiresInstanceReceiver)
+                        {
+                            Debug.Assert(visitedReceiver != null);
+                            var local = _factory.StoreToTemp(visitedReceiver, out var store, refKind: visitedReceiver.GetRefKind());
+                            visitedReceiver = local;
+                            locals.Add(local.LocalSymbol);
+                            sideEffectsBuilder.Add(store);
+                            receiverStored = true;
+                        }
+
+                        for (int index = lastIndexStored + 1; index <= lastNeededParameter; index++, lastIndexStored++)
+                        {
+                            var paramIndex = argsToParamsOpt.IsDefault ? index : argsToParamsOpt[index];
+                            RefKind argRefKind = argumentRefKindsOpt.RefKinds(index);
+                            RefKind paramRefKind = parameters[paramIndex].RefKind;
+
+                            Debug.Assert(!positionsAssignedTemps[index]);
+                            Debug.Assert(visitedArgumentsBuilder[index] != null);
+
+                            var visitedArgument = visitedArgumentsBuilder[index];
+                            var local = _factory.StoreToTemp(visitedArgument, out var store, refKind: paramRefKind == RefKind.None ? RefKind.None : argRefKind);
+
+                            visitedArgumentsBuilder[index] = local;
+                            locals.Add(local.LocalSymbol);
+                            sideEffectsBuilder.Add(store);
+                            positionsAssignedTemps[index] = true;
+
+                            if (refKindsBuilder != null)
+                            {
+                                refKindsBuilder[index] = GetPatchedRefKind(argRefKind, paramRefKind);
```

Comment:
>refKindsBuilder[index] = GetPatchedRefKind(argRefKind, paramRefKind) [](http://example.com/codeflow?start=32&length=68)

I am not sure why do we need to do this here. It doesn't look like the value is going to depend on anything that we do in this function.